pkg/extensionController/controller.go

```diff
@@ -209,7 +209,15 @@ func (c *controllerImpl) UninstallExtension(txCtx *transaction.TransactionContex
 	if err != nil {
 		return extensionLoadingFailed(extensionId, err)
 	}
-	return extension.Uninstall(c.createExtensionContext(txCtx), extensionVersion)
+	extensionCtx := c.createExtensionContext(txCtx)
+	instances, err := extension.ListInstances(extensionCtx, extensionVersion)
+	if err != nil {
+		return fmt.Errorf("failed to check existing instances: %w", err)
+	}
+	if len(instances) > 0 {
+		return fmt.Errorf("cannot uninstall extension because %d instance(s) still exist", len(instances))
+	}
+	return extension.Uninstall(extensionCtx, extensionVersion)
 }
 
```

Comment:
Only one questions: Who will receive this error in which situation?
Will it be an interactive (GUI) user seeing this error message?
Is there a mitigation EM could propose to the user? 
And does the user know what an "instance" is?

Reply:
Good point. Currently this would result in an internal server error (500). I should change this to a 400 status.
Actually the UI should not allow uninstalling as long as instances exist. So this is just additional safety.